drivers/usb/usbxhci/roothub.c

```diff
@@ -0,0 +1,185 @@
+/*
+ * PROJECT:     ReactOS USB xHCI Miniport Driver
+ * LICENSE:     GPL-2.0+ (https://spdx.org/licenses/GPL-2.0+)
+ * PURPOSE:     USBXHCI root hub functions
+ * COPYRIGHT:   Copyright 2023 Ian Marco Moffett <ian@vegaa.systems>
+ */
+
+#include "usbxhci.h"
+
+#define NDEBUG
+#include <debug.h>
+
+MPSTATUS
+NTAPI
+XHCI_RH_ChirpRootPort(IN PVOID XhciExtension,
+                      IN USHORT Port)
+{
+    DPRINT("XHCI_RH_ChirpRootPort: UNIMPLEMENTED. FIXME\n");
+    return MP_STATUS_SUCCESS;
+}
+
+VOID
+NTAPI
+XHCI_RH_GetRootHubData(IN PVOID XhciExtension,
+                       IN PVOID RootHubData)
```

Comment:
I would recommend for all functions where you had to use at least one line-break for the params list already (like here), that you do a line-break after the opening brace (after the functions name) already and do each parameter with just a 4 spaces indentation. That has the advantage that not all parameters indentation have to be realigned if any function would ever have to be renamed for any reason.

Furthermore it might make sense to use SAL2 annotations.

But all in all the formatting of this PR is surprisingly polished already. Good style!

Reply:
All right, will update that :+1: